AgentInterfaces/VmAgentSettings.cs

```diff
@@ -21,6 +21,12 @@ public class VmAgentSettings
         /// </summary>
         public HashSet<string> TitlesUsingExternalAllocations { get; set; }
 
+        /// <summary>
+        /// Titles that are using MaintenanceV2 - essentially, getting the full maintenance schedule details on the
+        /// GSDK maintenance callback
+        /// </summary>
+        public HashSet<string> TitlesUsingMaintenanceV2 { get; set; } 
```

Comment:
Instead of a thunderfig setting, shouldn't this be a flag that the GSDK sends to VmAgent? Like, as part of its first heartbeat can't it send a "usingMaintenanceV2" flag?